### PR TITLE
feat: Make response for different content types

### PR DESCRIPTION
**Pull request by laurensmertens**
|Q              |A  |
|---------------|---|
|Bug fix?       |No|
|New feature?   |Yes|	
|BC breaks?     |No|	
|Deprecations?  |No|	
|Fixed tickets  |No|	

Created a makeResponse function to allow any header and any content specified in the template.

### CODE REVIEW

**Review by Davidmattei — CHANGES_REQUESTED** [review]

File: Controller/RouterController.php [hunk 1]
```diff
@@ -58,4 +59,24 @@ public function asset(Request $request): Response
...
+
+        $response = new Response();
+
+        foreach ($data['headers'] as $key => $value) {
+            $response->headers->add([
+                $key => $value,
+            ]);
+        }
+
+        $response->setContent($data['content']);
```
[the remark this answers is not in the document]
Comment (laurensmertens) [reply Controller/RouterController.php]: @Davidmattei OK fixed

**Review by Davidmattei — DISMISSED** [review]

File: Controller/RouterController.php [hunk 2]
```diff
@@ -58,4 +59,24 @@ public function asset(Request $request): Response
 
         return $this->processor->getResponse($request, $data['hash'], $data['config'], $data['filename']);
     }
+
+    public function makeResponse(Request $request): Response
+    {
+        $result = $this->handler->handle($request);
+        $json = $this->templating->render($result['template'], $result['context']);
+
+        $data = \json_decode($json, true);
```
Comment (theus77) [inline Controller/RouterController.php:68]: The outcome of the json_encode must be tested (=== false?) and an error thrown.

File: Controller/RouterController.php [hunk 3]
```diff
@@ -58,4 +59,24 @@ public function asset(Request $request): Response
...
+
+        $response = new Response();
+
+        $response->setContent($data['content']);
```
Comment (theus77) [inline Controller/RouterController.php:72]: What if content is missing?

File: Controller/RouterController.php [hunk 4]
```diff
@@ -58,4 +59,24 @@ public function asset(Request $request): Response
...
+
+        foreach ($data['headers'] as $key => $value) {
```
Comment (theus77) [inline Controller/RouterController.php:74]: What if headers is missing?

File: Controller/RouterController.php [hunk 5]
```diff
@@ -58,4 +60,30 @@ public function asset(Request $request): Response
...
+
+    public function makeResponse(Request $request): Response
+    {
+        $result = $this->handler->handle($request);
+        $json = $this->templating->render($result['template'], $result['context']);
+
+        $data = \json_decode($json, true);
+
+        if ($data === false) {
+            throw new \RuntimeException('Failed to decode JSON');
```
Comment (theus77) [inline Controller/RouterController.php:72]: Th error message should explain that a json is expected with at least a content field string

File: Controller/RouterController.php [hunk 6]
```diff
@@ -58,4 +60,30 @@ public function asset(Request $request): Response
...
+        }
+
+        $response = new Response();
+
+        $response->setContent($data['content'] ?? null);
```
Comment (theus77) [inline Controller/RouterController.php:77]: If content is missing you should raise an error with a message explaining that a json with a mandatory content field is required

File: Controller/RouterController.php [hunk 7]
```diff
@@ -58,4 +60,35 @@ public function asset(Request $request): Response
...
+        $result = $this->handler->handle($request);
+        $json = $this->templating->render($result['template'], $result['context']);
+
+        $data = \json_decode($json, true);
+
+        if ($data === false) {
+            throw new \RuntimeException('JSON is expected with at least a content field as a string');
+        }
+
+        if (!array_key_exists('content', $data)) {
```
Comment (theus77) [inline Controller/RouterController.php:75]:
```suggestion
        if (!\is_string($data['content'] ?? null)) {
```

File: Controller/RouterController.php [hunk 8]
```diff
@@ -58,4 +60,35 @@ public function asset(Request $request): Response
...
+            throw new \RuntimeException('JSON requires at least a content field as a string');
+        }
+
+        $response = new Response();
+
+
+        $response->setContent($data['content']);
+
+        $headers = $data['headers'] ?? ['Content-Type' => 'text/plain'];
```
Comment (theus77) [inline Controller/RouterController.php:84]:
```suggestion
        $headers = $data['headers'] ?? ['Content-Type' => 'text/plain'];
        
        
        if  (!\is_array($headers)) {
                throw new \RuntimeException('Unexpected non-array headers parameter');
        }
```

**Review by theus77 — APPROVED** [review]

**Pull request merged**